### DISCUSSION TITLE
Ruby  ❤️ Programming Language Thread

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#51392] [Programming Help]
[Ruby](https://github.com/ruby/ruby) is a simple and elegant programming language that mimics natural syntax. _$ denotes a paid resource._ 
 
**Around the web**

- [Learn Ruby the Hard Way](https://learncodethehardway.org/ruby/)
- [Learning Ruby: A Resource List](https://github.com/fpsvogel/learn-ruby) 
- [The Odin Project - Full Stack Ruby on Rails](https://www.theodinproject.com/paths/full-stack-ruby-on-rails?) 
- $ [Learn Enough Ruby to Be Dangerous](https://www.learnenough.com/ruby) 
- [Exercism: Ruby](https://exercism.org/tracks/ruby) 

Books

- Head First Ruby: A Brain-Friendly Guide - Jay McGavren
- Ruby Best Practices: Increase Your Productivity - Write Better Code - Gregory T Brown
- The Well-Grounded Rubyist - David A. Black & Joseph Leo III 

Share any links, tutorials, videos, and books that have helped you learn Ruby below - don’t forget to include if it is paid too!

**Comment by sergeyleschev** [comment]
[Ruby Documentation](https://ruby-doc.org/)
[Try Ruby](https://ruby.github.io/TryRuby/)
[Ruby Koans](http://rubykoans.com/)
[RubyMonk](https://rubymonk.com/)
[Learn Ruby](https://www.codecademy.com/learn/learn-ruby)
YouTube:
[Ruby Programming Language - Full Course](https://www.youtube.com/watch?v=t_ispmWmdjY) - FreeCodeCamp
[Ruby Tutorial for Beginners](https://www.youtube.com/watch?v=Dji9ALCgfpM) - Programming with Mosh
[Ruby on Rails Crash Course](https://www.youtube.com/watch?v=fmyvWz5TUWg) - FreeCodeCamp

**Reply by queenofcorgis** [reply]
Thanks for sharing - we appreciate it!
